src/NuGet/Microsoft.Net.Compilers/CompilerArtifacts.props

```diff
@@ -11,40 +11,57 @@
     <_Exes>$(ArtifactsConfigurationDir)Exes\</_Exes>
   </PropertyGroup>
 
+  <!--
+    The CompilerArtifact list is used to generate Microsoft.Net.Compilers package, VS.Toolset.Roslyn CoreXT package and Microsoft.CodeAnalysis.Compilers Willow VSIX.
+    Keeping the list in central location allows us to make sure that these packages include the same files.
+
+    Ngen* attributes indicate how VS should NGEN the assemblies that are distributed by VS Setup.
+    NgenApplication is relative to the Roslyn install directory within MSBuild (i.e. MSBuild\15.0\Bin\Roslyn).
+  -->
+
   <ItemGroup>
     <!-- The Roslyn built binaries must be taken from these locations because this is the location where signing occurs -->
-    <CompilerArtifact Include="$(_Dlls)Microsoft.CodeAnalysis\Microsoft.CodeAnalysis.dll"/>
+    <CompilerArtifact Include="$(_Dlls)Microsoft.CodeAnalysis\Microsoft.CodeAnalysis.dll" NgenArchitecture="all" NgenApplication="VBCSCompiler.exe"/>
     <CompilerArtifact Include="$(_Dlls)Microsoft.CodeAnalysis\Microsoft.DiaSymReader.Native.amd64.dll"/>
     <CompilerArtifact Include="$(_Dlls)Microsoft.CodeAnalysis\Microsoft.DiaSymReader.Native.x86.dll"/>
-    <CompilerArtifact Include="$(_Dlls)Microsoft.CodeAnalysis.CSharp\Microsoft.CodeAnalysis.CSharp.dll"/>
-    <CompilerArtifact Include="$(_Dlls)Microsoft.CodeAnalysis.Scripting\Microsoft.CodeAnalysis.Scripting.dll"/>
-    <CompilerArtifact Include="$(_Dlls)Microsoft.CodeAnalysis.CSharp.Scripting\Microsoft.CodeAnalysis.CSharp.Scripting.dll"/>
-    <CompilerArtifact Include="$(_Dlls)Microsoft.CodeAnalysis.VisualBasic\Microsoft.CodeAnalysis.VisualBasic.dll"/>
-    <CompilerArtifact Include="$(_Exes)csc\net46\csc.exe"/>
+    <CompilerArtifact Include="$(_Dlls)Microsoft.CodeAnalysis.CSharp\Microsoft.CodeAnalysis.CSharp.dll" NgenArchitecture="all" NgenApplication="VBCSCompiler.exe"/>
+    <CompilerArtifact Include="$(_Dlls)Microsoft.CodeAnalysis.Scripting\Microsoft.CodeAnalysis.Scripting.dll" NgenArchitecture="all" NgenApplication="csi.exe"/>
+    <CompilerArtifact Include="$(_Dlls)Microsoft.CodeAnalysis.CSharp.Scripting\Microsoft.CodeAnalysis.CSharp.Scripting.dll" NgenArchitecture="all" NgenApplication="csi.exe"/>
+    <CompilerArtifact Include="$(_Dlls)Microsoft.CodeAnalysis.VisualBasic\Microsoft.CodeAnalysis.VisualBasic.dll" NgenArchitecture="all" NgenApplication="VBCSCompiler.exe"/>
+    <CompilerArtifact Include="$(_Exes)csc\net46\csc.exe" NgenArchitecture="all" NgenApplication="csc.exe" NgenPriority="1"/>
     <CompilerArtifact Include="$(_Exes)csc\net46\csc.exe.config"/>
     <CompilerArtifact Include="$(_Exes)csc\net46\csc.rsp"/>
-    <CompilerArtifact Include="$(_Exes)csi\net46\csi.exe"/>
+    <CompilerArtifact Include="$(_Exes)csi\net46\csi.exe" NgenArchitecture="all" NgenApplication="csi.exe" NgenPriority="1"/>
```

Comment:
Why is there a priority entry on only some of the elements here?

Reply:
Don't know. We haven't had it on others. I preserved the values we used.